commands/connect/mapping-write-errors.js

```diff
@@ -1,37 +1,36 @@
 'use strict'
 const api = require('../../lib/connect/api.js')
 const regions = require('../../lib/connect/regions.js')
+const pg = require('heroku-pg')
 const cli = require('heroku-cli-util')
 const co = require('co')
 
 function * run (context, heroku) {
   context.region = yield regions.determineRegion(context, heroku)
-  let mappingName = context.args.name
+  let mappingName = context.args.name.toLowerCase()
   let connection = yield api.withConnection(context, heroku)
-  let mapping = yield api.withMapping(connection, mappingName)
-  let results = yield cli.action('Retrieving errors', co(function * () {
-    let url = '/api/v3/mappings/' + mapping.id + '/errors'
-    return yield api.request(context, 'GET', url)
-  }))
-  let errors = results.json
+  let query = `
+SELECT
+  id, processed_at, record_id, sf_message, false AS archived
+FROM
+  ${connection.schema_name}._trigger_log
+WHERE
+  state = 'FAILED' AND table_name = '${mappingName}'
+UNION
+SELECT
+  id, processed_at, record_id, sf_message, true AS archived
+FROM
+  ${connection.schema_name}._trigger_log_archive
+WHERE
+  state = 'FAILED' AND table_name = '${mappingName}'
+ORDER BY
+  processed_at DESC
+`
+  let fetcher = pg.fetcher(heroku)
+  let db = yield fetcher.database(context.app, connection.db_key)
 
-  if (errors.count === 0) {
-    cli.log('No write errors in the last 24 hours')
-  } else {
-    if (context.flags.json) {
-      cli.styledJSON(errors.results)
-    } else {
-      cli.table(errors.results, {
-        columns: [
-          {key: 'id', label: 'Trigger Log ID'},
-          {key: 'table_name', label: 'Table Name'},
-          {key: 'record_id', label: 'Table ID'},
-          {key: 'message', label: 'Error Message'},
-          {key: 'created_at', label: 'Created'}
-        ]
-      })
-    }
-  }
+  let output = yield pg.psql.exec(db, query)
```

Comment:
I think its somewhat strange to access the trigger log directly and not via the API. This way you will need to upgrade both CLI tool and connect at the same time, since there is not abstraction layer.
I mean the trigger log is public API anyways, but still.

Reply:
There's no way to get anything older than 24 hours without the API. This provides users with *all* write errors.

> This way you will need to upgrade both CLI tool and connect at the same time, since there is not abstraction layer.

I don't understand your concern here. Can you share more about your concern here?

Reply:
I talked to a friend recently and we where discussing syncing IoT sensor data to Salesforce using Connect. Imagine they deploy a bug and millions of sensor requests error.

If your new API isn't limited or has a pg statement timeout, it is a possible attack vector. I attacker creates a demo connection. Inserts manually a couple million rows into the `_trigger_log` table, and blasts your API into oblivion. Murphy's Law 😜 
Never assume a sane user, we're all bananas 🍌